public/app/core/utils/file_export.ts

```diff
@@ -54,29 +54,46 @@ function formatRow(row, addEndRowDelimiter = true) {
   return addEndRowDelimiter ? text + END_ROW : text;
 }
 
-export function convertSeriesListToCsv(seriesList, dateTimeFormat = DEFAULT_DATETIME_FORMAT, excel = false) {
+export function convertSeriesListToCsv(
+  seriesList,
+  dateTimeFormat = DEFAULT_DATETIME_FORMAT,
+  excel = false,
+  removeNullValues = false
+) {
   let text = formatSpecialHeader(excel) + formatRow(['Series', 'Time', 'Value']);
   for (let seriesIndex = 0; seriesIndex < seriesList.length; seriesIndex += 1) {
     for (let i = 0; i < seriesList[seriesIndex].datapoints.length; i += 1) {
-      text += formatRow(
-        [
-          seriesList[seriesIndex].alias,
-          moment(seriesList[seriesIndex].datapoints[i][POINT_TIME_INDEX]).format(dateTimeFormat),
-          seriesList[seriesIndex].datapoints[i][POINT_VALUE_INDEX],
-        ],
-        i < seriesList[seriesIndex].datapoints.length - 1 || seriesIndex < seriesList.length - 1
-      );
+      if ((removeNullValues && seriesList[seriesIndex].datapoints[i][POINT_VALUE_INDEX]) || !removeNullValues) {
```

Comment:
this will filter out zero & false values as well. 

try
```js
const value = seriesList[seriesIndex].datapoints[i][POINT_VALUE_INDEX];
 if(removeNullValues && value === null) {
continue;
} 
```

and then reuse the value in the expression below

Reply:
nice, that's definitely cleaner - will update the code